src/Attributes/UnixTime.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+
+declare(strict_types=1);
+
+namespace Crell\Serde\Attributes;
+
+use Attribute;
+use Crell\AttributeUtils\SupportsScopes;
+use Crell\Serde\TypeField;
+
+#[Attribute(Attribute::TARGET_PROPERTY)]
+class UnixTime implements TypeField, SupportsScopes
+{
+    /**
+     * @param bool $milliseconds
+     *   Whether the field is represented as milliseconds (true) or seconds (false) since the epoch.
+     * @param array<string|null> $scopes
+     */
+    public function __construct(
+        public readonly bool $milliseconds = false,
```

Comment:
There's also microseconds, then the boolean breaks down. One could consider using an enum.

Reply:
My knee-jerk response here is to agree.  Something like `accuracy` or `resolution`, with values of seconds, milliseconds, or microseconds?

That said... Can DateTime even handle both mili and microseconds?  What time formats actually use those?

Reply:
So, I considered beyond millisecond resolution and decided against it for several reasons:

1. A Unix Timestamp is defined as **seconds** since the Unix Epoch.
2. However, JavaScript gives it in milliseconds and this is unlikely to change. 
3. Browsers and PHP servers kind of go together, hence the support of millisecond resolution. 
4. I have never, in all of my years writing PHP needed less than one millisecond of precision from timestamps sent over the wire. At that point, you are measuring scientific data which should probably be kept as an int instead of turned into a date (IMHO, but I could be persuaded otherwise). Dates are for humans in PHP, not computers. They offer some syntax for comparison, but ultimately, they are designed to be formatted and there is not a standard format for less than 1ms resolution. 
5. To my knowledge, Python is the only language that supports this resolution of Unix Timestamps via `time_ns`. I'm not sure why Python did this. I've never used it, but I've done quite a bit of Python over the years working with data scientists. 
6. Nanosecond/microsecond resolution appears to be a very niche case that a general library probably shouldn't solve.

That was my reasoning, but I am happy to implement it in whatever way makes sense.

Reply:
To play devils advocate to myself:

We can't always choose the format of the data we have to (de)serialize. The library should be flexible enough to handle it.

Reply:
I've updated this by adding a "resolution" enum: `UnixTimeResolution`.

Note that PHP crashes with nanosecond resolution. That means we truncate it to microsecond resolution when importing/exporting the value, even though libraries like Carbon can actually support nanosecond resolution.

Reply:
(also, we can't "round" to microsecond resolution because PHP totally butchers the rounding in a nonsensical way)